habitat/datasets/rearrange/samplers.py

```diff
@@ -525,6 +567,27 @@ def sample(
             # now find a matching link
             for link_ix in range(ao_instance.num_links):
                 if ao_instance.get_link_name(link_ix) == self.link_name:
+                    if receptacle is not None:
+                        if (
+                            ao_instance.handle
+                            == receptacle.parent_object_handle
+                        ) and (
+                            (link_ix == receptacle.parent_link)
+                            or ("frige" in receptacle.parent_object_handle)
+                            or ("fridge" in receptacle.parent_object_handle)
```

Comment:
I understand why this hardcoded override is necessary with current metadata, but I'm hesitant to merge it. I'd be more comfortable with a metadata solution which could be applied generically for other cases like this. Example: a receptacle inside the ReplicaCAD cabinet.

One idea, we could annotate receptacles with an additional optional field for link state requirements to access the space. In this paradigm, the fridge receptacles could be annotated with an acceptable joint range for the doors which allow access to them. Also note that this would be a very easy change since AO Receptacle metadata are defined with the URDF rather than distributed across the scene and stage configs.